test/unit/org/apache/cassandra/db/commitlog/CommitLogTest.java

```diff
@@ -614,34 +619,49 @@ protected Void testRecovery(byte[] logData, int version) throws Exception
         return null;
     }
 
-    protected Void testRecovery(CommitLogDescriptor desc, int endPosition, byte[] logData, boolean tolerateTruncation) throws Exception
+    protected Void testRecovery(CommitLogDescriptor desc, byte[] logData) throws Exception
     {
         File logFile = tmpFile(desc.version);
         CommitLogDescriptor fromFile = CommitLogDescriptor.fromFileName(logFile.getName());
         // Change id to match file.
         desc = new CommitLogDescriptor(desc.version, fromFile.id, desc.compression, desc.getEncryptionContext());
+        
         ByteBuffer buf = ByteBuffer.allocate(1024);
         CommitLogDescriptor.writeHeader(buf, desc, getAdditionalHeaders(desc.getEncryptionContext()));
-
-        // Save offset
-        // TODO: better descriptive name? "syncMarkerOffset"? "descriptorLength"?
-        int offset = buf.position();
         
-        // Write expected end position
-        // TODO: is there a more accurate/descriptive name for what this value represents?
-        buf.putInt(endPosition);
+        try (OutputStream lout = new FileOutputStream(logFile))
+        {
+            lout.write(buf.array(), 0, buf.position());
+            lout.write(logData);
```

Comment:
Nit: This can only work for `logData` being empty (since we don't support pre-3.0 commit logs, all segments must always include sync markers). Could we remove it?

Reply:
Ok, sure. 
I only restored that `testRecovery` method to the way it was b/c I thought it maybe better not to disturb the existing test (`testRecoveryWithBadCompressor`). I let that override the earlier decision to have both tests use the same method. I will put it back the way it was the first time :)